### PR TITLE
Computing message queue positions

### PR DESCRIPTION
**Pull request by yk**
Introduces counters for work queues that allows us to track the positions of enqueued work requests without having to iterate through the queues

### CODE REVIEW

File: inference/server/oasst_inference_server/routes/chats.py [hunk 1]
```diff
@@ -133,19 +134,33 @@ async def message_events(
...
+                item = await message_queue.dequeue(timeout=settings.pending_event_interval)
                 if item is None:
                     if not has_started:
+                        if work_queue is None:
+                            qpos, qlen = 0, 1
+                        else:
+                            # TODO: make more efficient, e.g. pipeline
+                            qdeq = await work_queue.get_deq_counter()
+                            qenq = await work_queue.get_enq_counter()
+                            mpos = await queueing.get_pos_value(redis_client, message_id)
```
Comment (AbdBarho) [inline inference/server/oasst_inference_server/routes/chats.py:157]: could this be done in parallel like this? or is this a race condition?

```python
[qdeq, qenq, mpos] = await asyncio.gather(
  work_queue.get_deq_counter(),
  work_queue.get_enq_counter(),
  queueing.get_pos_value(redis_client, message_id)
)
```

Comment (yk) [reply inference/server/oasst_inference_server/routes/chats.py]: good idea

File: inference/worker/utils.py [hunk 2]
```diff
@@ -76,6 +76,7 @@ def wait_for_inference_server(inference_server_url: str, timeout: int = 600):
 
 
 def text_to_events(text: str, seed: int | None = None, pause: float = 0.0):
+    time.sleep(2)
```
Comment (melvinebenezer) [inline inference/worker/utils.py:79]: did not understand how this helps in computing message queue positions. is this needed ?

**Review by melvinebenezer — APPROVED** [review]

File: inference/server/oasst_inference_server/queueing.py [hunk 3]
```diff
@@ -3,36 +3,76 @@
...
+        self.expire = expire
+        self.with_counter = with_counter
+        self.counter_pos_expire = counter_pos_expire
 
-    async def enqueue(self, value: str, expire: int | None = None) -> None:
-        pushed = await self.redis_client.rpush(self.queue_id, value)
-        if expire is not None:
-            await self.set_expire(expire)
-        return pushed
+    async def enqueue(self, value: str) -> None:
```
[the remark this answers is not in the document]
Comment (yk) [reply inference/server/oasst_inference_server/queueing.py]: thank you

**Pull request merged**